src/repl/scala/tools/nsc/interpreter/ExprTyper.scala

```diff
@@ -40,7 +40,28 @@ trait ExprTyper {
         case Success =>
           val sym0 = symbolOfTerm(name)
           // drop NullaryMethodType
-          sym0.cloneSymbol setInfo exitingTyper(sym0.tpe_*.finalResultType)
+          sym0.cloneSymbol setInfo exitingTyper {
+            val tp = sym0.tpe_*
+            if (tp.resultType.isTrivial && allowInterpretTwice) {
+              val origSym = symbolOfTerm(code)
+              if (origSym.isAccessor && !origSym.isMutableVal) {
```

Comment:
Does this mean this PR won't work for @NthPortal's `:type (23: 23)` test?
https://github.com/scala/scala/pull/8553/files#r349956405

Reply:
indeed, it won't work for that case; I didn't find a way to distinguish if `:type`'s input is a constant or not around related code.